docarray/typing/__init__.py

```diff
@@ -1,16 +1,22 @@
 from docarray.typing.id import ID
-from docarray.typing.tensor import NdArray, Tensor, TorchEmbedding, TorchTensor
+from docarray.typing.tensor import NdArray, Tensor
+
+try:
+    import torch  # noqa: F401
+except ImportError:
+    pass
+else:
+    from docarray.typing.tensor import TorchEmbedding, TorchTensor  # noqa: F401
+
 from docarray.typing.tensor.embedding import Embedding
 from docarray.typing.url import AnyUrl, ImageUrl, TextUrl
 
 __all__ = [
-    'TorchTensor',
     'NdArray',
     'Embedding',
     'ImageUrl',
     'TextUrl',
     'AnyUrl',
     'ID',
     'Tensor',
-    'TorchEmbedding',
 ]
```

Comment:
Can we add the `try` for the import at the bottom, then we can do:
```python
try:
    import torch  # noqa: F401
except ImportError:
    pass
else:
    from docarray.typing.tensor import TorchEmbedding, TorchTensor  # noqa: F401
    __all__.extend(['TorchEmbedding', 'TorchTensor'])
```

Reply:
> Can we add the `try` for the import at the bottom, then we can do:
> 
> ```python
> try:
>     import torch  # noqa: F401
> except ImportError:
>     pass
> else:
>     from docarray.typing.tensor import TorchEmbedding, TorchTensor  # noqa: F401
>     __all__.extend(['TorchEmbedding', 'TorchTensor'])
> ```

I had the same idea and I was wondering if this is good practice ? Probably okayish lets do it